Update to Datadog Disaster Recovery docs for GA launch at DASH 2026 (#36623)
DataDog/documentation: content/en/agent/guide/datadog-disaster-recovery.md

#### Create your DDR org

1. Go to [Get Started with Datadog](https://app.datadoghq.com/signup). You may need to log out of your current session, or use incognito mode to access this page.
2. Choose a different Datadog site than your primary (for example, if you're on `US1`, choose `EU` or `US5`).
3. Follow the prompts to create an account.

All Datadog sites are geographically separated. Reference the [Datadog Site List](https://docs.datadoghq.com/getting_started/site#access-the-datadog-site) for options.

If you are also sending telemetry to Datadog using cloud provider integrations, you must add your cloud provider accounts in the DDR org. Datadog does not use cloud providers to receive telemetry data while the DDR site is passive (not in failover).

#### Share the DDR org information with Datadog

Email your new org name to your [Customer Success Manager](mailto:success@datadoghq.com). Then, your Customer Success Manager sets this new org as your DDR org.

{{% /collapse-content %}}

{{% collapse-content title="Retrieve the public IDs and link your DDR and primary orgs" level="h5" %}}

For security reasons, Datadog is unable to link the orgs on your behalf.

> After the Datadog team has set your DDR org, use the Datadog [public API endpoint](https://docs.datadoghq.com/api/latest/organizations/#list-your-managed-organizations) to retrieve the public IDs of the primary and DDR org.

Reviewer: @nhamlani101 I assume Claude changed all the links from reference style (the numbers) to inline like this. However, our style guide specifies reference-style: https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#links

You might want to add the style guide to your Claude settings for making docs changes in the future so it knows not to change things like this.

#### Create your DDR org

1. Go to [Get Started with Datadog](https://app.datadoghq.com/signup). You may need to log out of your current session, or use incognito mode to access this page.
2. Choose a different Datadog site than your primary (for example, if you're on `US1`, choose `EU` or `US5`).
3. Follow the prompts to create an account.

All Datadog sites are geographically separated. Reference the [Datadog Site List](https://docs.datadoghq.com/getting_started/site#access-the-datadog-site) for options.

If you are also sending telemetry to Datadog using cloud provider integrations, you must add your cloud provider accounts in the DDR org. Datadog does not use cloud providers to receive telemetry data while the DDR site is passive (not in failover).

#### Share the DDR org information with Datadog

Email your new org name to your [Customer Success Manager](mailto:success@datadoghq.com). Then, your Customer Success Manager sets this new org as your DDR org.

> **Note:** Although this org appears in your Datadog billing hierarchy, all usage and cost associated is *not* billed during the Preview period.

Reviewer 2: Still mentioning the preview period, is that right? Also I'm about 75% certain that it doesn't show up in their billing hierarchy.

Author: Gah this stayed from the original doc, my bad, thanks for seeing this

Remote Configuration is enabled by default for new orgs, including your DDR org. Any new API keys you create are RC-enabled for use with your Agent. For more details, see the [Remote Configuration documentation][7].
[Remote Configuration (RC)][11] allows you to remotely configure and change the behavior of Datadog Agents deployed in your infrastructure.

Datadog strongly recommends using Remote Configuration for a more seamless failover control. As an alternative to RC, you can manually configure your Agents or use configuration management tools such as Puppet, Ansible, or Chef.
{{% /collapse-content %}}
Remote Configuration is enabled by default for new orgs, including your DDR org. Any new API keys you create are RC-enabled for use with your Agent. For more details, see the [Remote Configuration documentation][11].

Datadog strongly recommends using Remote Configuration for better failover control. As an alternative to RC, you can manually configure your Agents or use configuration management tools such as Puppet, Ansible, or Chef.

{{% collapse-content title="Dual Ship telemetry to DDR org during failover or drills" level="h5" %}}
{{% /collapse-content %}}

> [Dual Shipping][17] allows you to simultaneously route the same data to two different orgs, such as a primary and a failover org. Starting with Agent **v7.54+**, a new DDR configuration enables Datadog Agents to send {{< tooltip text="telemetry" tooltip="Data that is sent to the Datadog platform. For example, `logs`, `metrics`, `traces`. " >}} to the designated failover org when failover is triggered. 

Reviewer: @nhamlani101 I think it's not a bad idea to describe what dual shipping is ("allows you to simultaneously route the same data to two different orgs"); is there a reason you removed it?

> From the [Fleet Automation][13] page in your failover org, on the {{< ui >}}Configure Agents{{< /ui >}} tab, you can create a new failover policy or reuse an existing one, and apply it to your fleet of Agents. Soon after the policy is enabled, Agents begin dual-shipping telemetry to both the primary and DDR (failover) observability sites.

Reviewer: ```suggestion
From the [Fleet Automation][13] page in your failover org, on the {{< ui >}}Configure Agents{{< /ui >}} tab, you can create a failover policy or reuse an existing one, and apply it to your fleet of Agents. Soon after the policy is enabled, Agents begin dual-shipping telemetry to both the primary and DDR (failover) observability sites.
```

From the [Fleet Automation][13] page in your failover org, on the {{< ui >}}Configure Agents{{< /ui >}} tab, you can create a new failover policy or reuse an existing one, and apply it to your fleet of Agents. Soon after the policy is enabled, Agents begin dual-shipping telemetry to both the primary and DDR (failover) observability sites.

To create a failover policy, click on {{< ui >}}Create Failover Policy{{< /ui >}}.

{{< img src="/agent/guide/ddr/ddr-fa-policy.png" alt="Manage DDR policies" style="width:80%;" >}}

Then, follow the prompt to scope the hosts and telemetry (metrics, logs, traces) that you are required to failover.

{{< img src="/agent/guide/ddr/ddr-fa-policy-scope.png" alt="Scope the hosts and telemetry required to failover" style="width:80%;" >}}

**Note**: Cloud Integrations can only run in either your primary or DDR Datadog site, but not both at the same time, so failing them over will cease Cloud Integration data in your primary site. **During an integration failover, integrations run only in the DDR data center**. When no longer in failover, disable the failover policy to return integration data collection to the primary org.

> <div class="alert alert-danger">

Reviewer: @nhamlani101 is there a reason you wanted to drop the formatting here?

Current:

<img width="751" height="275" alt="Image" src="https://github.com/user-attachments/assets/9a6e3a50-35eb-4c20-8968-abdcfee4a134" />

With your change:

<img width="792" height="168" alt="Image" src="https://github.com/user-attachments/assets/195d3562-214b-4125-ad6c-edbf2524e931" />

> <div class="alert alert-danger">Cloud Integrations can only run in either your primary or DDR Datadog site, but not both at the same time, so failing them over will cease Cloud Integration data in your primary site. <strong>During an integration failover, integrations run only in the DDR data center.</strong> When no longer in failover, disable the failover policy to return integration data collection to the primary org.</div>

Reviewer: ```suggestion
<div class="alert alert-danger">Cloud Integrations can only run in either your primary or DDR Datadog site, but not both at the same time, so failing them over ceases Cloud Integration data in your primary site. <strong>During an integration failover, integrations run only in the DDR data center.</strong> When no longer in failover, disable the failover policy to return integration data collection to the primary org.</div>
```

Managed sync replicates user accounts from your primary org to your DDR org. Datadog recommends configuring [Just-in-Time provisioning with SAML](https://docs.datadoghq.com/account_management/saml/#just-in-time-jit-provisioning) to ensure your users can access the DDR org during a failover without needing to reset their password.

{{% /collapse-content %}}

{{% collapse-content title="Set up your cloud integrations (AWS, Azure, Google Cloud)" level="h5" %}}

See the [AWS](https://docs.datadoghq.com/integrations/amazon-web-services), [Azure](https://docs.datadoghq.com/integrations/azure), and [Google Cloud](https://docs.datadoghq.com/integrations/google-cloud-platform/?tab=organdfolderlevelprojectdiscovery#overview) integrations for setup steps.

Your cloud integrations must be configured in both primary and DDR orgs. Datadog ensures integrations run in only one org at a time: by default in the primary org, and in the DDR org during failover.

For more information, see the [Cloud integrations failover](#id-for-cloud) section.

{{% /collapse-content %}}

{{% collapse-content title="Set up credentials for managed resource sync" level="h5" id="syncing-data" %}}

> Datadog manages resource sync on your behalf using the open-source [datadog-sync-cli](https://github.com/DataDog/datadog-sync-cli) tool. You do not need to run or operate this tool yourself.

Reviewer: ```suggestion
Datadog manages resource sync on your behalf using the open source [datadog-sync-cli](https://github.com/DataDog/datadog-sync-cli) tool. You do not need to run or operate this tool yourself.
```

{{% /collapse-content %}}

{{% collapse-content title="Enable Remote Configuration [**RECOMMENDED]" level="h5" %}}

[Remote Configuration (RC)](https://docs.datadoghq.com/agent/remote_config/?tab=configurationyamlfile) allows you to remotely configure and change the behavior of Datadog Agents deployed in your infrastructure.

Remote Configuration is enabled by default for new orgs, including your DDR org. Any new API keys you create are RC-enabled for use with your Agent. For more details, see the [Remote Configuration documentation](https://docs.datadoghq.com/agent/remote_config/?tab=configurationyamlfile).

Datadog strongly recommends using Remote Configuration for better failover control. As an alternative to RC, you can manually configure your Agents or use configuration management tools such as Puppet, Ansible, or Chef.

{{% /collapse-content %}}

{{% collapse-content title="Dual ship telemetry to DDR org during failover or drills" level="h5" %}}

> **Dual Shipping is disabled by default**, but you can enable it to support your periodic disaster recovery exercises and drills.

Reviewer: ```suggestion
```

"dd-api-key:${PRIMARY_DD_API_KEY}" -H \
"dd-application-key:${PRIMARY_DD_APP_KEY}" --data "${CONNECTION}" --request POST ${PRIMARY_DD_API_URL}/api/v2/hamr
```

After linking your orgs, only the failover org displays this banner:

{{< img src="agent/guide/ddr/ddr-banner.png" alt="The DDR banner in the DDR org" >}}

{{% /collapse-content %}}

### 2. Set up access, integrations, syncing, and agents

{{% collapse-content title="Configure Single Sign On for the DDR org" level="h5" %}}

**Datadog recommends using Single Sign On (SSO)** to enable all your users to log in to your Disaster Recovery org during an outage.

Go to the [Organization Settings](https://app.datadoghq.com/organization-settings/users) in your DDR org to configure [SAML](https://docs.datadoghq.com/account_management/saml/#overview) or {{< ui >}}Google Login{{< /ui >}} for your users.

> You must invite each of your users to your Disaster Recovery org and give them appropriate roles and permissions. Alternatively, to streamline this operation, you can use [Just-in-Time provisioning with SAML](https://docs.datadoghq.com/account_management/saml/#just-in-time-jit-provisioning).

Reviewer 2: > You must invite each of your users to your Disaster Recovery org and give them appropriate roles and permissions.

I think we can get around this now with auto-verify and managed-sync.

{{% /collapse-content %}}

{{% collapse-content title="Set up your cloud integrations (AWS, Azure, Google Cloud)" level="h5" %}}

See the [AWS](https://docs.datadoghq.com/integrations/amazon-web-services), [Azure](https://docs.datadoghq.com/integrations/azure), and [Google Cloud](https://docs.datadoghq.com/integrations/google-cloud-platform/?tab=organdfolderlevelprojectdiscovery#overview) integrations for setup steps.

Your cloud integrations must be configured in both primary and DDR orgs. However, the integrations must only run in one org at a time:

- By default, the integrations must run only in the primary org.

> - When in failover, the integrations must only run in the DDR org.

Reviewer 2: The language here is weird to me that we keep telling them that their integrations "must" only run in one org at a time. They have no control over that, that's on us. If I were reading this doc as someone trying to implement this I might go looking for how to configure my integrations to run in only one org at a time.

For more information, see the [Cloud integrations failover](#id-for-cloud) section.

{{% /collapse-content %}}

{{% collapse-content title="Set up credentials for managed resource sync" level="h5" id="syncing-data" %}}

Datadog manages resource sync on your behalf using the open-source [datadog-sync-cli](https://github.com/DataDog/datadog-sync-cli) tool. You do not need to run or operate this tool yourself.

> Managed sync replicates resources from your primary org to your DDR org on a daily schedule. Replicated resources include dashboards, monitors, users, notebooks, and [34+ other resource types](https://github.com/DataDog/datadog-sync-cli#supported-resources). Replication runs continuously before an outage, so your DDR org stays current.

Reviewer 2: I would read "continuously" as different than "daily." Continuously means if I make a change in R1 it would be immediately reflected in R2.

> **Users are scoped to each Datadog site.** A user account in your primary org does not automatically exist in your DDR org. Datadog recommends configuring [Just-in-Time provisioning with SAML](https://docs.datadoghq.com/account_management/saml/#just-in-time-jit-provisioning) to help ensure your users can access the DDR org during a failover without manual user creation. You can also manually invite users to both orgs and assign them the same roles and permissions.

Reviewer 2: I think Managed sync + auto verify means we can copy their users....although the more I think about this I don't know how they'd get their password? They might have to do a reset password in R2? I'll try to find out.

> **Managed sync uses a Datadog [service account](https://docs.datadoghq.com/account_management/org_settings/service_accounts/).** During onboarding, you will need to create a service account in your primary org to read and replicate resources to your DDR org. Resources synced by managed sync appear under this service account in your DDR org rather than their original owner. This is expected behavior and does not affect resource functionality during failover.

Reviewer 2: >  will need to create a service account in your primary org 

The service account needs to be in the secondary/ddr org not the primary.

> Resources synced by managed sync appear under this service account in your DDR org rather than their original owner. This is expected behavior and does not affect resource functionality during failover.

"Resources synced by managed sync will be provisioned by a user mapped to their original owner when possible."

Remote Configuration is enabled by default for new orgs, including your DDR org. Any new API keys you create are RC-enabled for use with your Agent. For more details, see the [Remote Configuration documentation](https://docs.datadoghq.com/agent/remote_config/?tab=configurationyamlfile).

Datadog strongly recommends using Remote Configuration for better failover control. As an alternative to RC, you can manually configure your Agents or use configuration management tools such as Puppet, Ansible, or Chef.

{{% /collapse-content %}}

{{% collapse-content title="Dual ship telemetry to DDR org during failover or drills" level="h5" %}}

[Dual Shipping](https://docs.datadoghq.com/agent/configuration/dual-shipping/?tab=helm) allows you to simultaneously route the same data to two different orgs, such as a primary and a failover org. Starting with Agent **v7.54+**, a new DDR configuration enables Datadog Agents to send {{< tooltip text="telemetry" tooltip="Data that is sent to the Datadog platform. For example, `logs`, `metrics`, `traces`. " >}} to the designated failover org when failover is triggered.

> **Dual Shipping is disabled by default**, but you can enable it to support your periodic disaster recovery exercises and drills.

Reviewer 2: I know you didn't add this, but I think it should be removed. It seems to refer to the always on dual shipping instead of what we're specifically talking about here, which is a more targeted dual shipping setup with fa/rc during an agent based fail over.

Reviewer: Removed!

  site: <DDR_SITE>  # For example "site: us5.datadoghq.com" for a US5 site
  api_key: <DDR_SITE_API_KEY>
```

{{% /tab %}}
{{< /tabs >}}

{{% /collapse-content %}}

{{% collapse-content title="Configure DNS-based failover" level="h5" %}}

DNS-based failover is a complementary approach to Agent-based failover. Instead of configuring Agents with a secondary site endpoint, you configure all your data sources to send telemetry to a single Datadog-provided custom intake URL. During a failover event, Datadog updates the DNS record for that URL to redirect traffic from your primary site to your DDR site.

> **Note:** DNS failover is all-or-nothing. All telemetry sources using your custom endpoint cut over simultaneously. Dual shipping is not supported for DNS-based failover.

Reviewer 2: > Dual shipping is not supported for DNS-based failover.

It think we confirmed that it can be configured for this, are you just saying we're not going to support it?

Author: @heyronhay What do you think is this something we should support?

Reviewer 3: @michael-richey I'm confused about what you are thinking about - dual shipping isn't possible with DNS-based failover. Are you thinking of backfill?

Author: Michael is referring to the network traffic cloning test that we did, should we say thats a possibility?

Reviewer 3: I wouldn't bring it up - it only works for a subset of traffic (only user-submitted telemetry, and "internal" telemetry, like that submitted by integrations or logs2metrics, can't be captured the same way).

Reviewer: I'll remove this line from the note: "Dual shipping is not supported for DNS-based failover"

Thanks!

You can make similar changes with either the official Helm chart or Datadog Operator if you need to specify a custom configuration. Otherwise, you can pass the settings as environment variables:

```
DD_MULTI_REGION_FAILOVER_ENABLED=true
DD_MULTI_REGION_FAILOVER_FAILOVER_METRICS=true
DD_MULTI_REGION_FAILOVER_FAILOVER_LOGS=true
DD_MULTI_REGION_FAILOVER_FAILOVER_APM=true
DD_MULTI_REGION_FAILOVER_SITE=ADD_NEW_ORG_SITE
DD_MULTI_REGION_FAILOVER_API_KEY=ADD_NEW_SITE_API_KEY
```

{{% /tab %}}
{{< /tabs >}}

{{% /collapse-content %}}

{{% collapse-content title="Activate and test DDR failover in cloud integrations" level="h5" id="id-for-cloud" %}}

You can test failover for your cloud integrations from your DDR organization's landing page.

{{< img src="/agent/guide/ddr/ddr-failover-main-page.png" alt="Enable the failover policy in the DDR org" style="width:80%;" >}}

On the failover landing page, you can check the status of your DDR org, or click {{< ui >}}Fail over your integrations{{< /ui >}} to test your cloud integration failover.

When no longer in failover, **disable the failover policy** in the DDR org to return integration data collection to the primary org.

> During testing, integration telemetry is spread over both organizations. If you cancel a failover test, the integrations return to running in the primary data center.

Reviewer 3: Do we want to go deeper into the reason for activating the integrations failover? How it only applies to integration crawlers, but not other integrations? It is an important distinction, but it may be tricky figuring out how to describe it.

Reviewer: Hey Ron, in the interest of getting this merged before DASH I'm going to skip this suggestion, but if you think it's something we should add in the future, would you mind leaving a comment on this followup ticket I made for myself? https://datadoghq.atlassian.net/browse/DOCS-14658

Thank you!
